### ISSUE TITLE
Is it possible write data to Azure SQL server from powershell? 

### ISSUE THREAD

**Issue by lhp620** [Azure/azure-powershell#6689]
I know some samples to write data to local SQL server, like:
https://nocolumnname.blog/2017/09/12/importing-excel-into-sql-server-using-powershell/ 

But I did not find a way to write data into Azure SQL server. If we can do this, could you please provide some sample codes?

**Comment by maddieclayton** [comment]
@jaredmoo Could you provide some assistance to this customer?

@lhp620, for reference, the Azure PowerShell SQL reference documentation can be found here: https://docs.microsoft.com/en-us/powershell/module/azurerm.sql/?view=azurermps-6.5.0

**Comment by lhp620** [comment]
Thank you. Please let me know it is possible to bulk import the data to Azure SQL server, like the way we use for local SQL server, like
	$BulkCopy = New-Object -TypeName Data.SqlClient.SqlBulkCopy -ArgumentList $ServerConnection;
	$BulkCopy.DestinationTableName = $Table;
	$BulkCopy.WriteToServer($DataTable);

if this is not possible, we can close this question now. 

I found an ugly way to import data by query command to Azure SQL server: 
"Install-Module -Name SqlServer -AllowClobber" to support Invoke-Sqlcmd and tried the following Powershell script, it works:

$params = @{
    'Database' = '<DB name>'
    'ServerInstance' = '<Azure SQL server name>'
    'Username' = '<username>'
    'Password' = '<password>'
    'Query' = $query
}
Invoke-Sqlcmd @params

**Comment by sqlmadhadr** [comment]
First off, thank you @lhp620 for the motivation to actually stand up a blog. I have asked myself the same question in the past and ended up writing a set of functions - I had to move 100+ tables over 1,000 times and needed to be able to schedule this event. Sometimes asking why isn't important 🥇 .

https://sqlmadhadr.blogspot.com/2018/07/using-powershell-to-bulk-copy-data.html

In this post I've provided SQLFn-ConnectionStrings.ps1, SQLFn-BulkInsert.ps1, and some examples of how one might use it. Maybe sometime soon I'll actually start building my own repo for this sort of stuff. Let me know if this helps or if I can answer any questions.
